apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerParameterCollection.cs

```diff
@@ -254,16 +254,16 @@ private void FillSpannerInternalValues(MapField<string, Value> valueDictionary,
             foreach (var parameter in _innerList)
             {
                 valueDictionary[GetCorrectedParameterName(parameter.ParameterName)]
-                    = parameter.SpannerDbType.ToProtobufValue(parameter.GetValidatedValue(), options);
+                    = parameter.GetConfiguredSpannerDbType(options).ToProtobufValue(parameter.GetValidatedValue(), options);
```

Comment:
Do we definitely need ToProtobufValue to still accept an options if it's part of GetConfiguredSpannerDbType?

The answer may well be yes - but now is the time to check.

Reply:
`ToProtobufValue` doesn't need options. I found that out the hard way that once we reach this method, its already late to make any type configuration changes based on options. Most of the tests and few call sites pass null for options.

Reply:
Yes, its internal and we can remove it . Contrary to YAGNI, I was just wondering if there may be a scenario in future where we may need to do some processing in this method :) 
Would you like this change to be part of this PR?

Reply:
Yes please - again, it's a matter of "get all the infrastructure to where we want it".

I think if we've got a fair amount of flexibility within SpannerDbType, and that's what we call ToProtobufValue on, I think it should be fine to remove it.

Reply:
Sure Jon.

Reply:
Done.